src/H5FDs3comms.c

```diff
@@ -1812,10 +1815,11 @@ H5FD__s3comms_load_aws_creds_from_file(FILE *file, const char *profile_name, cha
                 strncpy(setting_pointers[setting_i], (const char *)line_buffer, strlen(line_buffer));
 
                 /* "trim" tailing whitespace by replacing with null terminator*/
-                buffer_i = 0;
-                while (!isspace(setting_pointers[setting_i][buffer_i]))
-                    buffer_i++;
-                setting_pointers[setting_i][buffer_i] = '\0';
+                end = strlen(line_buffer) - 1;
```

Comment:
Be careful that the length of `line_buffer` isn't 0 here, which would cause `end` to wrap around.

Reply:
This should be covered by the earlier check on line 1810 that throws an errors if ` *(line_buffer + 1) == 0`.